fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/api/resource/ResourceManagerHelper.java

```diff
@@ -52,4 +54,27 @@ default void addReloadListener(IdentifiableResourceReloadListener listener) {
 	static ResourceManagerHelper get(ResourceType type) {
 		return ResourceManagerHelperImpl.get(type);
 	}
+
+	/**
+	 * Registers a built-in resource pack.
+	 *
+	 * <p>A built-in resource pack is an extra resource pack provided by your mod which is not always active, it's similar to the "Programmer Art" resource pack.
+	 *
+	 * <p>Why and when to use it? A built-in resource pack should be used to provide extra assets/data that should be optional with your mod but still directly provided by it.
+	 * For example it could provide textures of your mod in another resolution, or could allow to provide different styles of your assets.
+	 *
+	 * <p>The {@code subPath} corresponds to a path in the JAR file which points to the resource pack folder. For example the subPath can be {@code "resourcepacks/extra"}.
+	 *
+	 * <p>Note about the enabled by default parameter: a resource pack cannot be enabled by default, only data packs can.
+	 * Making this work for resource packs is near impossible without touching how Vanilla handles disabled resource packs.
+	 *
+	 * @param id The identifier of the resource pack.
```

Comment:
How should this id be namespaced? If I'm just dropping in a zip into my folder, I don't have to specify an id, just the filename identifies it as far as I'm aware. How would I translate that filename to a valid id?

Reply:
ZIP file? in which folder?

There's no ZIP file related.

The id is visible to the user, the goal of making this namespaced is to avoid any potential conflict.
Though the id does not influence the path at all.

Reply:
My resource packs are all zips (and so are the datapacks which I've created so far). It's great that this is all obvious to you, I'm just giving you my perspective as a newcomer and potential user of this API.

Reply:
Mods provides a resource pack which root is the JAR file (which allows MC to read `<mod JAR>/assets` and `<mod JAR>/data`

So not all resource packs are zips, you can have a folder resource pack, virtual resource pack, etc.

Reply:
This is my reference, and all I needed to create and work with resource packs: https://minecraft.gamepedia.com/Resource_Pack

Hence I know about using plain folders, which is okay for testing but you'd never distribute them that way. I'm glad that you're answering my questions, but it would be even more helpful if the documentation had reflected this knowledge in the first place.